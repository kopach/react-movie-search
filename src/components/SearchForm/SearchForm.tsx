import React from 'react'
import Card from 'react-bootstrap/Card'
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'

const SearchForm: React.FC<ISearchForm> = ({ title, year }): JSX.Element => (
  <Card>
    <Card.Body>
      <Form action="search" method="get" autoComplete="off">
        <Form.Group controlId="movieTitle">
          <Form.Label>Title</Form.Label>
          <Form.Control name="t" type="text" placeholder="Title" defaultValue={title} />
        </Form.Group>

        <Form.Group controlId="movieYear">
          <Form.Label>Year</Form.Label>
          <Form.Control name="y" type="number" placeholder="Year" defaultValue={year} />
        </Form.Group>

        <Button variant="outline-primary" type="submit" block>
          Search
        </Button>
        <Button variant="outline-secondary" type="button" block disabled>
          I'm Feeling Lucky
        </Button>
      </Form>
    </Card.Body>
  </Card>
)

export { SearchForm }

interface ISearchForm {
  title?: string
  year?: string
}
