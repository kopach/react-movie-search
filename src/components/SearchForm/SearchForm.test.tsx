import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { SearchForm } from './SearchForm'

describe('<SearchForm />', () => {
  let component: ReturnType<typeof shallow>

  beforeEach(() => {
    component = shallow(<SearchForm />)
  })

  it('should mount', () => {
    expect(component.length).toBe(1)
  })

  it('renders as expected', () => {
    const tree = renderer.create(<SearchForm />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
