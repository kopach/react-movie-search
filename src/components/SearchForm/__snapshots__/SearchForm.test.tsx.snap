// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> renders as expected 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <form
      action="/search"
      autoComplete="off"
      className=""
      method="get"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="movieTitle"
        >
          Title
        </label>
        <input
          className="form-control"
          id="movieTitle"
          name="t"
          placeholder="Title"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="movieYear"
        >
          Year
        </label>
        <input
          className="form-control"
          id="movieYear"
          name="y"
          placeholder="Year"
          type="number"
        />
      </div>
      <button
        className="btn btn-outline-primary btn-block"
        disabled={false}
        type="submit"
      >
        Search
      </button>
      <button
        className="btn btn-outline-secondary btn-block"
        disabled={true}
        type="button"
      >
        I'm Feeling Lucky
      </button>
    </form>
  </div>
</div>
`;
